engine/apps/alerts/models/alert_group.py

```diff
@@ -401,15 +401,25 @@ def is_alert_a_resolve_signal(self, alert):
         raise NotImplementedError
 
     @property
-    def permalink(self):
+    def slack_permalink(self):
         if self.slack_message is not None:
             return self.slack_message.permalink
 
+    @property
+    def telegram_permalink(self) -> typing.Optional[str]:
+        from apps.telegram.models.message import TelegramMessage
+
+        main_telegram_message = self.telegram_messages.filter(
```

Comment:
Please check if "telegram_messages" should be added to prefetch_related (probably using django's `Prefetch` to filter out only needed messages?).

Reply:
hmm.. it's a tough call as I'm not yet very familiar with how `prefetch_related` works in Django. It actually looks like by adding it to `prefetch_related` it results in 1 additional SQL query?

See below from some local test results when fetching two alert groups that both have a discord message associated with them.

Without `telegram_messages` in `IncidentSerializer.PREFETCH_RELATED`:
![without](https://user-images.githubusercontent.com/9406895/191792355-8ab46d96-cdbe-4045-9b9c-1071da93c740.png)

With `telegram_messages` in `IncidentSerializer.PREFETCH_RELATED`:
![with](https://user-images.githubusercontent.com/9406895/191792371-d21458b1-23ad-407a-91bc-e666292341c5.png)

Reply:
I'll try scripting a quick way to generate a bunch of alert groups and telegram messages. I think the difference may become evident then with a large amount of data.

Reply:
hm.. from there, it seems the prefetched data is not being used, since you still have the 2 queries to get the telegram messages (maybe because the data is not used in the serializer instance but inside a called method? the idea behind prefetch is to get all the rows you will need in one query, and then use that prefetched data instead of triggering new queries in this case you will be introducing an extra/unused query that could be expensive).

Reply:
@matiasb those screenshots are a bit outdated. @vadimkerr and I paired for a bit and it looks like the following results in only one additional query to `telegram_telegrammessage`: 
```python3
PREFETCH_RELATED = [
        "alerts",
        Prefetch(
            "telegram_messages",
            TelegramMessage.objects.filter(chat_id__startswith="-", message_type=TelegramMessage.ALERT_GROUP_MESSAGE),
        ),
    ]
```